pkg/metrics/metrics.go

```diff
@@ -0,0 +1,129 @@
+package metrics
+
+import (
+	"time"
+
+	"github.com/prometheus/client_golang/prometheus"
+	"sigs.k8s.io/controller-runtime/pkg/metrics"
+)
+
+var (
+	upStatus = prometheus.NewGauge(prometheus.GaugeOpts{
+		Name: "up_status",
+		Help: "The up status of the build-controller.",
+	})
+
+	buildCount = prometheus.NewGaugeVec(
+		prometheus.GaugeOpts{
+			Name: "build_count",
+			Help: "Build creation count.",
+		},
+		[]string{"build"})
+
+	buildRunCount = prometheus.NewGaugeVec(
+		prometheus.GaugeOpts{
+			Name: "buildrun_count",
+			Help: "BuildRun execution count.",
+		},
+		[]string{"buildrun"})
+
+	buildRunEstablishDuration = prometheus.NewGaugeVec(
+		prometheus.GaugeOpts{
+			Name: "buildrun_establish_duration",
+			Help: "BuildRun establish duration in seconds.",
+		},
+		[]string{"buildrun"})
+
+	buildRunReadyDuration = prometheus.NewGaugeVec(
+		prometheus.GaugeOpts{
+			Name: "buildrun_ready_duration",
+			Help: "BuildRun ready duration in seconds.",
+		},
+		[]string{"buildrun"})
+
+	buildRunReadyDurationHis = prometheus.NewHistogramVec(
+		prometheus.HistogramOpts{
+			Name:    "buildrun_ready_duration_his",
+			Help:    "BuildRun ready duration histogram in seconds.",
+			Buckets: prometheus.LinearBuckets(50, 50, 10),
+		},
+		[]string{"buildrun"})
+)
+
+//CustomBuildMetrics This struct implements the Collector interface to be registered in the middleware
+//by the Prometheus server
+type CustomBuildMetrics struct {
+	UpStatus                      prometheus.Gauge
+	BuildCount                    prometheus.GaugeVec
+	BuildRunCount                 prometheus.GaugeVec
+	BuildRunEstablishDuration     prometheus.GaugeVec
+	BuildRunReadyDuration         prometheus.GaugeVec
+	BuildRunReadyDurationHis      prometheus.HistogramVec
+}
+
+//ProxyMetrics of the package level item.
```

Comment:
leave empty space between `//` and the comment.

Reply:
fixed